torch/csrc/jit/import_source.cpp

```diff
@@ -91,88 +93,104 @@ struct SourceResolver : public Resolver {
       Function& m,
       const SourceRange& loc) const override {
     auto it = env_.find(name);
-    if (it == env_.end()) {
-      return nullptr;
+    if (it != env_.end()) {
+      return it->second;
     }
-    return it->second;
+
+    if (imports_.count(name)) {
+      return std::make_shared<ClassNamespaceValue>(name);
+    }
+    return nullptr;
   }
 
   TypePtr resolveType(const std::string& name) const override {
     return ClassType::get(name);
   }
 
  private:
+  std::unordered_set<std::string> imports_;
   std::unordered_map<std::string, std::shared_ptr<SugaredValue>> env_;
 };
 
-// Helper that contains the state for a parsing a TorchScript source string.
-struct SourceImporter {
-  SourceImporter(
-      const std::string& src,
-      const std::vector<at::Tensor>& constant_table)
-      : parser_(src) {
-    const auto version = parseVersionNumber();
-    resolver_ = std::make_shared<SourceResolver>(version, constant_table);
-  }
-
-  Parser parser_;
-  ResolverPtr resolver_;
+static size_t parseVersionNumber(Lexer& L) {
+  auto range = L.cur().range;
+  auto name = L.expect(TK_IDENT).text();
+  L.expect('=');
+  std::string version_text = L.expect(TK_NUMBER).text();
+  L.expect(TK_NEWLINE);
+  auto version = Const::create(L.cur().range, version_text);
+  if (name != "op_version_set")
+    throw ErrorReport(range) << "expected an assignment to op_version_set";
+  if (!version.isIntegral())
+    throw ErrorReport(range)
+        << "expected an integral version but found " << version.text();
+  return size_t(version.asIntegral());
+}
 
-  size_t parseVersionNumber() {
-    auto& L = parser_.lexer();
-    auto range = L.cur().range;
-    auto name = L.expect(TK_IDENT).text();
-    L.expect('=');
-    std::string version_text = L.expect(TK_NUMBER).text();
+static std::unordered_set<std::string> parseImports(Lexer& L) {
+  std::unordered_set<std::string> imports;
```

Comment:
unordered_set will cause non-deterministic behavior.  This should use a vector and another thing should  avoid loading duplicates.